### ISSUE TITLE
socket.join doesn't exist

### ISSUE THREAD

**Issue by sabrehagen** [fullstackers/socket.io-events#18]
In vanilla socket.io, `join` exists on the socket object (see http://socket.io/docs/server-api/#socket#join(name:string[,-fn:function]):socket). When calling `join` on a `socket` in the function with signature `function (socket, args, next)`, it is reported that `socket.join` is not a function. This means existing socket.io code is not directly compatible with socket.io-events. Can you add the join feature to the sockets for these functions please?

**Comment by sabrehagen** [comment]
also, `socket.broadcast.*` doesn't work as expected.

**Comment by NathanGRomano** [comment]
I am working on this.  The `socket` is a wrapper that delegates calls to the real socket.  It is possible to do `socket.sock.join`.  However, that just feels dirty.  So I will be working those issues out.  Thanks for bringing this up.  This gives me motivation to update it.

**Comment by sabrehagen** [comment]
Thanks. That's great news. I transformed my whole socket.io code base to the express architecture as enabled by your project, and it instantly became far more expressive. It's such a massive improvement on the standard routing library, and a perfect integration with an existing express codebase. Unfortunately this was the only limiting factor so I had to postpone the conversion. Very much looking forward to having it :)

On another topic, what's your opinion on the format of the `args` parameter? It feels a little hacky to me. I'm finding myself putting at the head of every function

```
var event = args[0];
var data = args[1];
var callback = args[2];
```

What do you do when using the library?

**Comment by NathanGRomano** [comment]
Well I currently do as you do right there.  I have been meaning to get support for acknowledgements and support for parameters.

**Comment by NathanGRomano** [comment]
I recently left my full time position with Aver Informatics.  I had owned a company called Turbonetix IO that I gave up to join another company and that failed.  So I moved the open source projects over to Full Stacker LLC.  This is a new company that I am working on full time right now.  There is a lot of work that my partner and I need to get done before we can commit to making significant changes to our projects.  We really need to schedule 8 hours a week to updating these projects.  I have so many things for this, and we really need to get ready for ES6.
